src/backends/odbc/standard-use-type.cpp

```diff
@@ -212,14 +211,20 @@ void odbc_standard_use_type_backend::pre_use(indicator const *ind)
     SQLSMALLINT sqlType(0);
     SQLSMALLINT cType(0);
     SQLLEN size(0);
+    SQLLEN bufLen(0);
 
     void* const sqlData = prepare_for_bind(size, sqlType, cType);
-
+    if (sqlType == SQL_LONGVARCHAR)
```

Comment:
Because with SQL_LONGVARCHAR it still have issue.
I've checked on both 10 on AppVeyor, and 11 - locally on Win 10, sql express 2016 - all works, fine as you can see.

Another option come into my mind - operates only with sizeof and move check for long ```size > 8000``` after ```prepare_for_bind```.

Anyway it solves the long standing issue - even if you don't like it.